sycl/doc/extensions/experimental/sycl_ext_oneapi_matrix/sycl_ext_oneapi_matrix.asciidoc

```diff
@@ -0,0 +1,663 @@
+
+## Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification section 6.3.3 "Feature test macros".  Therefore, an
+implementation supporting this extension must predefine the macro
+`SYCL_EXT_ONEAPI_MATRIX` to one of the values defined in the table below.
+Applications can test for the existence of this macro to determine if the
+implementation supports this feature, or applications can test the macro's
+value to determine which of the extension's APIs the implementation supports.
+
+[frame="none",options="header"]
+|======================
+|Value |Description
+|1     |The APIs of this experimental extension are not versioned, so the feature-test macro always has this value. 
+|======================
+
+## Matrix API Versions
+
+While this document presents the core API that unifies Intel AMX, DPAS, and Nvidia Tensor Cores, the implementations support slightly different versions of the API. For this reason, we introduce a new macro, namely `SYCL_EXT_ONEAPI_MATRIX_VERSION`  to distinguish between these different implementations. The goal in the next few months is to get rid of this implementation versioning macro. These are the current values for this macro.
+
+[frame="none",options="header"]
+|======================
+|Value |Description
+|1     |Initial extension JIT implementation on Intel AMX and DPAS. load, store, mad, fill, piece-wise operations, and the query interface are supported. The old API used for this implementation is detailed in [matrix extension](doc/extensions/deprecated/sycl_ext_oneapi_deprecated_matrix_no_use.asciidoc)
+|2     |JIT implementation on Intel AMX and DPAS. load, store, mad, fill, piece-wise operations, and the query interface are supported 
+|3     |Implementation on Nvidia Tensor Cores
+|======================
+
+## New `joint_matrix` class
+We introduce a new class called `joint_matrix`. The user needs to specify the type of the elements, shape, the matrix use, the memory layout, and the memory scope of the matrix. This results in the following description:
+
+```c++
+namespace sycl::ext::oneapi::experimental::matrix {
+template <typename T, use Use,
+          size_t Rows=sycl::dynamic_extent, size_t Cols=sycl::dynamic_extent,
+          layout Layout = layout::dynamic, typename Group = sub_group>
+struct joint_matrix {
+    joint_matrix(Group g) {}
+};
+}
+```
+
+IMPORTANT: Matrix layout defaulting to `layout::dynamic` applies only to matrix with `use::accumulator`
```

Comment:
There is no default for use::a and use::b. the default value `dynamic` applies only to matrix::accumulator. 
In the commit comment, by "restrict the API in this document to only the portable API", I meant you can only write portable code with this documented API.

Is the note in the text good enough?
"IMPORTANT: Matrix layout defaulting to `layout::dynamic` applies only to matrix with `use::accumulator`"

Reply:
> In the commit comment, by "restrict the API in this document to only the portable API", I meant you can only write portable code with this documented API.

If you still want to support `layout::dynamic` for `use::a` and `use::b`, then you have a bug in the definition of `joint_matrix_load` because that function only allows loading a matrix with dynamic layout for `use::accumulator`.

Reply:
It is the portable API, so it should not allow `layout::dynamic` for `use::a` and `use::b`.
This case will be added to the intel-specific extension